src/components/ChecklistPanel.js

```diff
@@ -86,6 +87,33 @@ class ChecklistPanel extends Component {
 		if ( prevProps.isPublishSidebarEnabled !== current && ! current ) {
 			this.props.onEnablePublishSidebar();
 		}
+		// Toggle the post saving lock.
+		this.toggleSavingLock();
+	}
+
+	toggleSavingLock() {
+		const {
+			onLockPostSaving,
+			onUnlockPostSaving,
+			shouldBlockPublish,
+		} = this.props;
+
+		if ( ! shouldBlockPublish ) {
+			return;
+		}
+
+		const { completion } = this.state;
+		const {
+			completed,
+			toComplete,
+		} = completion;
+		const isCompleted = completed >= toComplete;
```

Comment:
Seems strange to unpack these just for the one line, and doesn't make it much more readable, could we condense?
```suggestion
		const isCompleted = completion.completed >= completion.toComplete;
```

Reply:
sure, it was your code originally 😛

Reply:
I believe it 😅